src/lib/ares_expand_name.c

```diff
@@ -120,24 +126,20 @@ int ares__expand_name_validated(const unsigned char *encoded,
                                 int alen, char **s, long *enclen,
                                 int is_hostname)
 {
-  int len, indir = 0;
+  int len, nlen, indir = 0;
   char *q;
   const unsigned char *p;
-  union {
-    ares_ssize_t sig;
-     size_t uns;
-  } nlen;
 
-  nlen.sig = name_length(encoded, abuf, alen, is_hostname);
-  if (nlen.sig < 0)
+  nlen = name_length(encoded, abuf, alen, is_hostname);
+  if (nlen < 0 || nlen >= SIZE_MAX)
```

Comment:
I'm not aware of a single architecture where `int` is larger than `size_t`, so this seems to check for something that cannot possibly happen?